internal/distro/fedora31/distro.go

```diff
@@ -44,7 +39,104 @@ type output struct {
 const Distro = common.Fedora31
 const ModulePlatformID = "platform:f31"
 
-func New() (*Fedora31, error) {
+type Fedora31 struct {
+	arches        map[string]arch
+	outputs       map[string]output
+	buildPackages []string
+}
+
+type Fedora31Arch struct {
+	name   string
+	distro *Fedora31
+	arch   *arch
+}
```

Comment:
Having two structs representing architecture in each of the distros is weird. I'd merge them. Can be done in a follow-up, though.

Reply:
Totally, these patches only add things. Have a massive cleanup PR to do on top.